packages/esm-patient-chart-app/src/visit/hooks/useServiceQueue.tsx

```diff
@@ -0,0 +1,70 @@
+import useSWRImmutable from 'swr/immutable';
+import { FetchResponse, openmrsFetch, useConfig, openmrsObservableFetch } from '@openmrs/esm-framework';
+import { ChartConfig } from '../../config-schema';
+import { Observable } from 'rxjs';
+
+export interface QueueServiceInfo {
+  uuid: string;
+  display: string;
+  name: string;
+  description: string;
+}
+
+export interface QueueEntryPayload {
+  visit: { uuid: string };
+  queueEntry: {
+    status: { uuid: string };
+    priority: { uuid: string };
+    queue: { uuid: string };
+    patient: { uuid: string };
+    startedAt: Date;
+  };
+}
+
+export function useServices(location: string) {
+  const apiUrl = `/ws/rest/v1/queue?location=${location}`;
+  const { data } = useSWRImmutable<{ data: { results: Array<QueueServiceInfo> } }, Error>(apiUrl, openmrsFetch);
```

Comment:
```suggestion
  const config = useConfig() as ChartConfig;
  const apiUrl = `/ws/rest/v1/queue?location=${location}`;
  
  const { data } = useSWRImmutable<{ data: { results: Array<QueueServiceInfo> } }, Error>( config.showServiceQueueFields && location ? apiUrl : null, openmrsFetch);
```

So that the request doesn't occur if `showServiceQueueFields` is falsy and no `location` is provided.